internal/testing/require/require_test.go

```diff
@@ -50,3 +52,620 @@ func TestCapturePanic(t *testing.T) {
+	struct2 := &testStruct{"hello"}
+
+	tests := []struct {
+		name        string
+		require     func(TestingT)
+		expectedLog string
+	}{
+		{
+			name: "Contains passes on contains",
+			require: func(t TestingT) {
+				Contains(t, "hello cat", "cat")
+			},
+		},
+		{
+			name: "Contains fails on empty",
+			require: func(t TestingT) {
+				Contains(t, "", "dog")
+			},
+			expectedLog: `expected "" to contain "dog"`,
+		},
+		{
+			name: "Contains fails on not contains",
+			require: func(t TestingT) {
+				Contains(t, "hello cat", "dog")
+			},
+			expectedLog: `expected "hello cat" to contain "dog"`,
+		},
+		{
+			name: "Contains fails on not contains with format",
+			require: func(t TestingT) {
+				Contains(t, "hello cat", "dog", "pay me %d", 5)
+			},
+			expectedLog: `expected "hello cat" to contain "dog": pay me 5`,
+		},
+		{
+			name: "Equal passes on equal: string",
+			require: func(t TestingT) {
+				Equal(t, "wazero", "wazero")
+			},
+		},
+		{
+			name: "Equal passes on equal: []byte",
+			require: func(t TestingT) {
+				Equal(t, []byte{1, 2, 3, 4}, []byte{1, 2, 3, 4})
+			},
+		},
+		{
+			name: "Equal passes on equal: struct",
+			require: func(t TestingT) {
+				Equal(t, &testStruct{name: "takeshi"}, &testStruct{name: "takeshi"})
+			},
+		},
+		{
+			name: "Equal fails on nil: string",
+			require: func(t TestingT) {
+				Equal(t, "wazero", nil)
+			},
+			expectedLog: `expected "wazero", but was nil`,
+		},
+		{
+			name: "Equal fails on nil: []byte",
+			require: func(t TestingT) {
+				Equal(t, []byte{1, 2, 3, 4}, nil)
+			},
+			expectedLog: `expected []byte{0x1, 0x2, 0x3, 0x4}, but was nil`,
+		},
+		{
+			name: "Equal fails on nil: struct",
+			require: func(t TestingT) {
+				Equal(t, &testStruct{name: "takeshi"}, nil)
+			},
+			expectedLog: `expected &require.testStruct{name:"takeshi"}, but was nil`,
+		},
+
+		{
+			name: "Equal fails on not same type: string",
+			require: func(t TestingT) {
+				Equal(t, "wazero", uint32(1))
+			},
+			expectedLog: `expected "wazero", but was uint32(1)`,
+		},
+		{
+			name: "Equal fails on not same type: []byte",
+			require: func(t TestingT) {
+				Equal(t, []byte{1, 2, 3, 4}, "wazero")
+			},
+			expectedLog: `expected []uint8([1 2 3 4]), but was string(wazero)`,
+		},
+		{
+			name: "Equal fails on not same type: struct",
+			require: func(t TestingT) {
+				Equal(t, &testStruct{name: "takeshi"}, "wazero")
+			},
+			expectedLog: `expected *require.testStruct(&{takeshi}), but was string(wazero)`,
+		},
+		{
+			name: "Equal fails on not equal: string",
+			require: func(t TestingT) {
+				Equal(t, "wazero", "walero")
+			},
+			expectedLog: `expected "wazero", but was "walero"`,
+		},
+		{
+			name: "Equal fails on not equal: uint64", // ensure we don't use multi-line output!
+			require: func(t TestingT) {
+				Equal(t, uint64(12), uint64(13))
+			},
+			expectedLog: `expected 12, but was 13`,
+		},
+		{
+			name: "Equal fails on not equal: []byte",
```

Comment:
I added long form comparisons for structs and binary while keeping short for normal things like errors, numbers and strings.